### PR TITLE
[synthetics] Update docs to reflect the changes to `expect` 

### LINKED ISSUE #3206: [Synthetics] Update docs to reflect the change to `expect` from @playwright/test
### Description

The [current docs](https://www.elastic.co/guide/en/observability/current/synthetics-create-test.html#synthetics-assertions-methods) refer to the `expect` methods, but in https://github.com/elastic/synthetics/pull/812, Synthetics has been moved over to `@playwright/test`'s `expect.  The docs should be updated to reflect this.

### Resources

Playwright docs are at https://playwright.dev/docs/test-assertions but @vigneshshanmugam may need to provide some guidance here.

### Collaboration

TBD. The docs and product team will work together to determine the best path forward.

### Point of contact.

**Main contact:** @vigneshshanmugam 

**Stakeholders:**
- @vigneshshanmugam 
- @paulb-elastic

### PR DESCRIPTION
**Pull request by colleenmcginnis**
Closes #3206

Currently the docs list the supported [Jest `expect` methods](https://www.elastic.co/guide/en/observability/current/synthetics-create-test.html#synthetics-assertions-methods). As of [@elastic/synthetics v1.5.0](https://github.com/elastic/synthetics/releases/tag/v1.5.0), we use Playwright Test `expect` assertions. This replaces the list of supported Jest methods with links to the Playwright assertion docs.

👋  @vigneshshanmugam I'm not sure if I've captured everything correctly.

cc @paulb-elastic

### CODE REVIEW

File: docs/en/observability/synthetics-create-test.asciidoc [hunk 1]
```diff
@@ -155,42 +155,43 @@ For more details on getting started with the Synthetics Recorder, refer to <<syn
...
 
 [discrete]
 [[synthetics-make-assertions]]
 = Make assertions
 
-Check that a value meets a specific condition.
 A more complex `step` might wait for a page element to be selected
 and then make sure that it matches an expected value.
 
+Elastic Synthetics uses `@playwright/test`'s `expect` function to make assertions and
```
Comment (vigneshshanmugam) [inline docs/en/observability/synthetics-create-test.asciidoc:191]: Would call this unsupported one as snapshot based assertions.  `except for the unsupported ones [Snapshot Assertions](https://playwright.dev/docs/api/class-snapshotassertions) toHaveScreenshot and toMatchSnapsho`

**Review by vigneshshanmugam — APPROVED** [review]
LGTM, Thanks @colleenmcginnis

File: docs/en/observability/synthetics-create-test.asciidoc [hunk 2]
```diff
@@ -155,42 +155,43 @@ For more details on getting started with the Synthetics Recorder, refer to <<syn
...
+supports all https://playwright.dev/docs/test-assertions[Playwright assertions] _except_
+`toHaveScreenshot` and `toMatchSnapshot`.
+
```
Comment (colleenmcginnis) [inline docs/en/observability/synthetics-create-test.asciidoc:194]: @vigneshshanmugam I'm not sure I understand your comment. Is this clearer and still correct?

>Elastic Synthetics uses `@playwright/test`'s `expect` function to make assertions and supports most [Playwright assertions](https://playwright.dev/docs/test-assertions). Elastic Synthetics does _not_ support [`toHaveScreenshot`](https://playwright.dev/docs/api/class-locatorassertions#locator-assertions-to-have-screenshot-1) or any [Snapshot Assertions](https://playwright.dev/docs/api/class-snapshotassertions).

Comment (vigneshshanmugam) [reply docs/en/observability/synthetics-create-test.asciidoc:194]: Sorry should have been eloborate. yeah I was just asking if we can link in to that Snapshot based assertions. Looks good to me.

Comment (colleenmcginnis) [reply docs/en/observability/synthetics-create-test.asciidoc:194]:
```suggestion
Elastic Synthetics uses `@playwright/test`'s `expect` function to make assertions
and supports most https://playwright.dev/docs/test-assertions[Playwright assertions].
Elastic Synthetics does _not_ support https://playwright.dev/docs/api/class-locatorassertions#locator-assertions-to-have-screenshot-1[`toHaveScreenshot`]
or any https://playwright.dev/docs/api/class-snapshotassertions[Snapshot Assertions].

```

**Review by vigneshshanmugam — APPROVED** [review]

**Review by bmorelli25 — APPROVED** [review]

**Pull request merged**